executor/executor_distsql.go

```diff
@@ -294,7 +302,28 @@ func closeAll(objs ...Closeable) error {
 	return nil
 }
 
-// XSelectIndexExec represents XAPI select index executor without result fields.
+// XSelectIndexExec represents DistSQL select index executor.
+// There are two execution modes. one is single-read, in which case we only need to read index keys.
+// Another one is double-read, we first do index request to get handles, we use each partial result to build
+// a lookupTableTask.
+//
+// Each lookupTableTask works like XSelectTableExec. it sorts the handles, sends an *tipb.SelectRequest, then
+// gets distsql.SelectResult which returns multiple distsql.PartialResults, we fetche all the rows from
+// each distsql.PartialResult, then sorts the rows by the original index order.
+//
+// So there would be many tasks built from index request, each task do its own table request.
```

Comment:
1. would ->might
2. each task do its own table request -> and each task does its own table request.